docs/diga_api.md

```diff
@@ -0,0 +1,216 @@
+
+## Summary
+
+- user enters prescription code in the DiGA (app)
+- insurance information is extracted from the code
+- a request is made to the api of the insurance to verify the code
+  - response contains `Tag der Leistungserbringung` which is the day a user started to use the app
+  - this is required for the billing process
+- another request is made for reimbursing the code
+
+## The prescription code
+
+Patients receive 16-character prescription codes by their insurances. The codes are used to activate the DiGA apps and for creating invoices for billing. The structure of the code is defined [here](https://www.gkv-datenaustausch.de/media/dokumente/leistungserbringer_1/digitale_gesundheitsanwendungen/technische_anlagen_aktuell_7/DiGA_Anlage_1_Technische_Anlage_zur_RL_V1.1_20210225.pdf) on page 9 (German).
+
+| Krankenkassencode/Kostenträgerkürzel | Version     | Unique code for insurance | Checksum    |
+| ------------------------------------ | ----------- | ------------------------- | ----------- |
+| 2 characters                         | 1 character | 12 characters             | 1 character |
+
+The first two characters are an identifier for the insurance. They can be used to get additional information from a mapping (xml) file which can be downloaded [here](https://kkv.gkv-diga.de/).
+
+The other parts contain a version (`A`) and a unique code for the patient & insurance. The last character is a checksum which is described [here](https://www.gkv-datenaustausch.de/media/dokumente/leistungserbringer_1/digitale_gesundheitsanwendungen/technische_anlagen_aktuell_7/Anhang3_Freischaltcode_Berechnungsregel_Pruefziffer_V1.0.pdf) with an open source implementation for different languages on [Github](https://github.com/bitmarck-service).
+
+Codes for testing can be downloaded as xlsx [here](https://www.gkv-datenaustausch.de/media/dokumente/leistungserbringer_1/digitale_gesundheitsanwendungen/technische_anlagen_aktuell_7/Pseudo-Codes.xlsx). Some of these codes are invalid with error codes defined [here](https://www.gkv-datenaustausch.de/media/dokumente/leistungserbringer_1/digitale_gesundheitsanwendungen/technische_anlagen_aktuell_7/DiGA_Anhang5_Fehlerausgaben_V1.0.1_20210423.pdf).
+
+| Rückgabewert | Freischaltcode   | Fehlertext                    | Erläuterung                                                                                                                                                                            |
+| ------------ | ---------------- | ----------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| 0            | 77AAAAAAAAAAAAAX | Anfrage gültig                | Aktuelles Tagesdatum für "Tag_der_Leistungserbringung" aus EDFC0-basis_2.0.0.xsd verweden. Für die DiGANr laut aus EDFC0-basis_2.0.0.xsd sind die letzten 3-Stellen mit 000 anzugeben. |
```

Comment:
I think we should content to English and provide the German original as footnote